jsapp/js/account/organization/OrganizationSettingsRoute.tsx

```diff
@@ -1,7 +1,156 @@
+import Button from 'jsapp/js/components/common/button';
+
+// Stores, hooks and utilities
+import useWhenStripeIsEnabled from 'js/hooks/useWhenStripeIsEnabled.hook';
+import {OrganizationUserRole, useOrganizationQuery} from 'js/account/organization/organizationQuery';
+import subscriptionStore from 'js/account/subscriptionStore';
+import envStore from 'js/envStore';
+import {getSimpleMMOLabel} from './organization.utils';
+
+// Constants and types
+import {ORGANIZATION_TYPE_SELECT_OPTIONS} from 'js/account/accountFieldsEditor.component';
+
+// Styles
+import styles from 'js/account/organization/organizationSettingsRoute.module.scss';
+
+interface State {
+  name: string;
+  website?: string;
+  type?: string;
+}
+
+/**
+ * Renders few fields with organization related settings, like name or website
+ * (with some logic in regards to their visibility). If user has necessary role,
+ * they can edit available fields.
+ */
 export default function OrganizationSettingsRoute() {
+  const orgQuery = useOrganizationQuery();
+  const [subscriptions] = useState(() => subscriptionStore);
+  const [state, setState] = useState<State>({name: ''});
+  const [isStripeEnabled, setIsStripeEnabled] = useState(false);
+
+  useEffect(() => {
+    if (orgQuery.data) {
+      setState({name: orgQuery.data.name});
+    }
+  }, [orgQuery.data]);
+
+  useWhenStripeIsEnabled(() => {
+    setIsStripeEnabled(true);
+  }, []);
+
+  const isUserAdminOrOwner = (
+    orgQuery.data?.request_user_role &&
+    [OrganizationUserRole.admin, OrganizationUserRole.owner]
+      .includes(orgQuery.data?.request_user_role)
+  );
+
+  const isPendingOrgPatch = orgQuery.data && orgQuery.isPending;
+
+  function handleSave() {
+    // TODO: call the API endpoint
+    // to be done while doing: https://www.notion.so/kobotoolbox/Add-react-query-mutation-hook-for-org-changes-1307e515f6548010b5d3c087b634f01a
+    console.log('save');
+  }
+
+  function handleChangeName(name: string) {
+    setState((prevState) => {return {...prevState, name};});
+  }
+
+  function handleChangeWebsite(website: string) {
+    setState((prevState) => {return {...prevState, website};});
+  }
+
+  function isNameValueValid(currentName: string) {
+    return !currentName;
+  }
+
+  function isWebsiteValueValid(currentWebsite: string) {
+    return !currentWebsite;
+  }
+
+  function getTypeLabel(typeName: string) {
+    // TODO: see if this would be an actual source of the organization type label
+    // to be done while doing: https://www.notion.so/kobotoolbox/Add-react-query-mutation-hook-for-org-changes-1307e515f6548010b5d3c087b634f01a
+    const foundLabel = ORGANIZATION_TYPE_SELECT_OPTIONS.find((item) => item.value === typeName)?.label;
+    return foundLabel || typeName;
+  }
+
+  const mmoLabel = getSimpleMMOLabel(
+    envStore.data,
+    subscriptionStore.activeSubscriptions[0],
+    false,
+    true
+  );
+  const mmoLabelLowercase = mmoLabel.toLowerCase();
+
+  if (!orgQuery.data) {
```

Comment:
I believe we should use the loading flag instead.

```suggestion
  if (orgQuery.data.isLoading) {
```

Reply:
Likely yes, but depends how you would handle query errors.

Reply:
Good idea, but you've probably meant `orgQuery.isLoading` :)